wafer_space/projects/tasks.py

```
@@ -90,53 +90,448 @@ def _extract_filename_from_url(url: str) -> str:
+        logger: Logger instance
+    """
+    logger.error("System failure detected: %s", error_summary)
+    check.fail(error_summary)
+    logger.info(
+        "  Check marked as FAILED (retry %d/%d available)",
+        check.retry_count,
+        check.max_retries,
+    )
+
+
+def _handle_exception_retry(check_id, exc, task_instance, logger):
```

Ruff (0.14.5)

Unused function argument: `task_instance`

(ARG001)